File: engine/engine-model/src/main/scala/tools/packet/alliance/AllianceMemberOnline.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class AllianceMemberOnline(private var _allianceId: Int, private var _guildId: Int, private var _characterId: Int, private var _online: Boolean) extends PacketInput {
  def allianceId: Int = _allianceId

  def guildId: Int = _guildId

  def characterId: Int = _characterId

  def online: Boolean = _online

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/AllianceMemberOnline.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record AllianceMemberOnline(Integer allianceId, Integer guildId, Integer characterId,
                                   Boolean online) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
